cardano-db-sync/src/Cardano/DbSync/Cache.hs

```diff
@@ -0,0 +1,37 @@
+module Cardano.DbSync.Cache
+    ( Cache
+    , newEmptyCache
+    , queryMAWithCache
+    ) where
+
+import           Control.Monad.IO.Class (MonadIO (liftIO))
+import           Control.Monad.Trans.Reader
+import           Data.ByteString.Internal (ByteString)
+import           Data.Map.Strict
+import qualified Data.Map.Strict as Map
+import           Data.IORef (IORef, modifyIORef, newIORef, readIORef)
+
+import           Cardano.Ledger.Mary.Value (AssetName (..))
+
+import qualified Cardano.Db as DB
+
+import           Database.Persist.Postgresql (SqlBackend)
+
+newtype Cache = MACache (IORef (Map (ByteString, AssetName) DB.MultiAssetId))
```

Comment:
Why are we using `((ByteString, AssetName)` tuple there instead of the fingerprint?

Reply:
The fingerprint needs to be computed by hashing these two, so it takes some time.